aws/route53_key_signing_key_test.go

```diff
@@ -0,0 +1,87 @@
+package aws
+
+import (
+	"context"
+	"sync"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+// Route 53 Key Signing Key can only be enabled with KMS Keys in specific regions,
+
+// testAccRoute53KeySigningKeyRegion is the chosen Route 53 Key Signing Key testing region
+//
+// Cached to prevent issues should multiple regions become available.
+var testAccRoute53KeySigningKeyRegion string
+
+// testAccProviderRoute53KeySigningKey is the Route 53 Key Signing Key provider instance
+//
+// This Provider can be used in testing code for API calls without requiring
+// the use of saving and referencing specific ProviderFactories instances.
+//
+// testAccPreCheckRoute53KeySigningKey(t) must be called before using this provider instance.
+var testAccProviderRoute53KeySigningKey *schema.Provider
+
+// testAccProviderRoute53KeySigningKeyConfigure ensures the provider is only configured once
+var testAccProviderRoute53KeySigningKeyConfigure sync.Once
+
+// testAccPreCheckRoute53KeySigningKey verifies AWS credentials and that Route 53 Key Signing Key is supported
+func testAccPreCheckRoute53KeySigningKey(t *testing.T) {
+	testAccPartitionHasServicePreCheck(route53.EndpointsID, t)
```

Comment:
I wonder if we shouldn't pull this over into the `ParallelTest()` anonymous `PreCheck` function. If later we add a `testAccPartitionHasServicePreCheck` linter, this would fail even though it shouldn't.